### PR TITLE
Freshness metrics

### PR DESCRIPTION
**Pull request by KludgeKML**
Adds a freshness metric for OS Places postcodes. Every hour we update the oldest postcode sourced from OS Places APi and allow prometheus to collect its age rounded down to the day. We can then see a graph of the age in seconds of the least-recently updated postcode. Since we're expecting the entire database to be refreshed roughly once a week, we should see this hover around the 7 mark. If it increases significantly above that, we may be seeing problems.

We will evaluate this for a while before deciding if we need to add an alarm state to it.

https://trello.com/c/8P4LRqvl/2171-investigate-how-we-can-handle-os-places-api-being-down

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

**Review by 1pretz1 — COMMENTED** [review]
Nice idea adding a metric! Left a couple comments on the implementation.

File: lib/collectors/global_prometheus_collector.rb
Comment (KludgeKML) [reply lib/collectors/global_prometheus_collector.rb:24]: I think this is pretty safe because the only way these records get updated is via OS Places API calls (with one exception: bulk CSV import, which is also a reasonable consideration of freshness and is unlikely to happen again anyway).

File: lib/collectors/global_prometheus_collector.rb
Comment (KludgeKML) [reply lib/collectors/global_prometheus_collector.rb:13]: It is simpler, but probably quite a bit less meaningful. Days since last update of oldest postcode will tell us if there are problems with individual postcodes getting missed somehow, whereas the most recent time only really gives us problems with OS Places API as an entity. It's a faster metric, definitely, and easier to understand, but I don't think those outweigh the benefits of the slower metric's focus on overall freshness of data (and since our aim here anyway is to have a slower metric and less sensitive alert system, perhaps not in the spirit of the ticket).

Comment (1pretz1) [reply lib/collectors/global_prometheus_collector.rb:13]: That makes sense, thanks for the explanation. Tracking the oldest does give you an extra catch of if a bug creeps in where some postcodes are not being updated 👍 Could we can reword the description to something a bit clearer, maybe "Days since oldest postcode was last updated via OS Places API" (could also add in the expected value e.g ~7 days).

Also notice that the current graph states 12 days, could this be a result of the bug introduced, or does this need looking into?

Comment (KludgeKML) [reply lib/collectors/global_prometheus_collector.rb:13]: The current graph on integration might be down to integration having a slower refresh rate, but yeah, I noticed that. I think once we merge this in and have it running in prod for a day or two we should take a look if it's not around 7 days.

File: lib/collectors/global_prometheus_collector.rb
Comment (1pretz1) [reply lib/collectors/global_prometheus_collector.rb:24]: For now yes, agreed. However, if the code changes in the future and the records are updated outside of OS places requests then this is metric may no longer be meaningful. I think relying on the standard timestamp values for harvesting data will probably trip someone up in the future. Don't think I'd classify this as a blocker as we're only at risk of breaking a metric but is a risk and one we'd probably want to consider more if we plan to alert on it 👍

File: lib/collectors/global_prometheus_collector.rb
Comment (KludgeKML) [reply lib/collectors/global_prometheus_collector.rb]: Message updated: "Days since oldest postcode was last updated via OS Places API (Expected in prod: ~7)"

**Review by 1pretz1 — APPROVED** [review]
LGTM

File: lib/collectors/global_prometheus_collector.rb
Comment (1pretz1) [reply lib/collectors/global_prometheus_collector.rb]: Awesome, thanks!

**Pull request merged**
